### ISSUE TITLE
ReifyBuckets error when using DISTINCT in the array builder opertator

### ISSUE THREAD

**Issue by dugdun** [precog/quasar#3216]
**Quasar Versions noticed in**: 27.1.0 and 27.1.9 was working in an earlier version but I forget which one it was originally created and ran against. The query was written against data in MongoDB and not tested against any other connector. I was just trying out the operator as I hadn't seen it before John's talk at the Backend SlamJam session a few weeks back.

**Query**:

```
select
        discipline
    ,   year
    ,   event
    ,   gender
    ,   [distinct country ...] as countriesWinningMedals
from
        `/mongodb-3.4/data/olympics`
group by
        discipline
    ,   year
    ,   gender
    ,   event
order by
        discipline
    ,   year
    ,   gender
    ,   event
```

**Error**:

<img width="712" alt="screen shot 2017-11-30 at 15 39 19" src="https://user-images.githubusercontent.com/828514/33458756-abacc6ba-d5e4-11e7-9e85-1590ad7086f6.png">

**Expected Results**: Array item with a distinct list of countries that won a medal for each grouping. This might not be a valid query, but it was working in earlier versions. As far as I know there are no customers using the array builder operator (`[fieldname ...]`), so we can state that `distinct` is invalid in this context. The query could be rewritten to use a sub-select to get the distinct values and then added to the array in the outer `SELECT` statement.

**Comment by alissapajer** [comment]
@wemrysi will this be fixed by what you're working on now?

**Comment by wemrysi** [comment]
@alissapajer @dugdun I think that might actually be fixed in `27.1.10`.

**Comment by wemrysi** [comment]
Oh, actually maybe not, I'd have to try it to see, I was thinking `ReifyAutoJoins`.
